src/Dispatch.purs

```diff
@@ -18,24 +27,53 @@ to share query types.
 -- primitives. In any given primitive, case on your specific query type, and return all
 -- others wrapped in Emit.
 
-data Dispatch item o a
+data Dispatch item o e a
```

Comment:
We're forced to carry this `e` around everywhere because it's the effect type necessary for the `SearchState` and thus the `SearchInput` render function. This is because the search state carries an effect around with it due to the implementation of the debouncer. It's a bit annoying because it's a heavy cost to carry another type parameter every single place `Dispatch` is used.

However, I found [this code from @garyb](https://github.com/slamdata/purescript-halogen/issues/479#issuecomment-331702420) that may work; this would obviate the need to carry this parameter everywhere.

Reply:
Yeah, I was thinking about this ... I think the other option would be passing `Dispatch item o e a` around along with `item` (for `Container`) ... I guess `SearchState` would still require `e` though, but in general it looks like each thing doesn't need access to all four parameters, but mostly just a way to create a `Dispatch item o e a`.

Or give up and use IO :trollface: